plugins/auth-backend/src/lib/OAuthProvider.ts

```diff
@@ -39,14 +40,40 @@ export type Options = {
   tokenIssuer: TokenIssuer;
 };
 
+const readState = (stateString: string): OAuthState => {
+  try {
+    const state = Object.fromEntries(
+      new URLSearchParams(decodeURIComponent(stateString)),
+    );
+    return {
+      nonce: state.nonce ?? '',
```

Comment:
Best not assign any empty strings here, it's essentially makes these optional values but without type checking.

I think it makes sense to check if both `state.nonce` and `state.env` are present and non-empty strings here, and throw appropriate errors otherwise.